src/halo_regions.jl

```diff
@@ -1,93 +1,76 @@
-import GPUifyLoops: @launch, @loop, @unroll
-
-const PeriodicBC = BoundaryCondition(Periodic, nothing)
-const NoFluxBC = BoundaryCondition(Flux, 0)
-
-function fill_halo_regions!(grid, field_tuples...)
-    for ft in field_tuples
-        field, fbcs, data = ft
-        fill_halo_region!(grid, Val(field), fbcs, data)
-    end
+# Halo filling algorithm for arbitrary flux, gradient, and value boundary conditions.
+# For these we fill halos so that no-fluxes are added to source terms when the PDE / 
```

Comment:
Would it make sense to dispatch on a no-flux boundary condition here?

Reply:
No --- because this is the boundary condition we use for `Gradient` and `Value` conditions as well. We could dispatch on a `Union` of the three types (`Flux`, `Gradient`, and `Value`), but I think it will be easier to maintain the code if we think of this as "default", and the other boundary conditions (`Periodic`, `NoPenetration`) as special cases.